doc/sphinx/source/esmvalcore/recipe.inc

```diff
@@ -1,5 +1,177 @@
+
+Note that the author, project, and references will need to be included in the
+config-references.yml file. The author name uses the format:
+last name (clipped after 4 letters) underscore first name (clipped to two
+letters) and all letters are lowercase. For instance, Mickey Mouse would be:
+mous_mi. Also note that this username is unlikely to be the same as the github
+user name.
+
+
+
+Datasets
+========
+
+The datasets section includes:
+
+- dataset name
+- Project (CMIP5 or 6, observations...)
+- experiment (historical/ RCP8.5 etc...)
+- Ensemble member
+- The time range
+- The model grid, gn or gr, (CMIP6 only).
+
+For example, a datasets section could be:
+
+.. code-block:: yaml
+
+    datasets:
+      - {dataset: CanESM2, project: CMIP5, exp: historical, ensemble: r1i1p1, start_year: 2001, end_year: 2004}
+      - {dataset: UKESM1-0-LL, project: CMIP6, exp: historical, ensemble: r1i1p1f2, start_year: 2001, end_year: 2004, grid: gn}
+
+
+Note that this section is not required, as datasets can also be provided in the
+`Diagnostics`_ section.
+
+
+Preprocessors
+=============
+
+The preprocessor section of the recipe includes one or more preprocesors, each
+of which may call one or several preprocessor functions.
+
+Each preprocessor section includes:
+
+- A preprocessor name.
+- A list of preprocesor functions to apply
+- Any Arguments given to the preprocessor functions.
+- The order that the preprocesor functions are applied can also be specified.
+
+The following preprocessor is an example of a preprocessor that contains
+multiple preprocessor functions:
+
+.. code-block:: yaml
+
+    preprocessors:
+      prep_timeseries: # For regional volume averaged (at the surface)
+        custom_order: true
+        extract_region:
+          start_longitude: -80.
+          end_longitude: 30.
+          start_latitude: -80.
+          end_latitude: 80.
+        extract_volume:
+          z_min: 0.
+          z_max: 100.
+        volume_statistics:
+          operator: mean
+        multi_model_statistics:
+          span: overlap
+          statistics: [mean ]
+
+If only the default preprocessor is needed, then this section can be omitted.
+
+
+Diagnostics
+===========
+
+The diagnostics section includes one or more
+diagnostics. Each diagnostics will have:
+
+- A list of which variables to load
+- A description of the variables (optional)
+- Which preprocessor to apply to each variable
+- The script to run
+- The diagnostics can also include an additional_datasets section, if specific
+  datasets are linked to specific diagnostics.
+
+The following example, taken from recipe_ocean_amoc.yml, shows a diagnostic
+named `diag_timeseries_amoc`, which calculates a derived variable and
+sends it to the ocean time series scipt.
+
+.. code-block:: yaml
+
+    diagnostics:
+      diag_timeseries_amoc:
+        description: atlantic_meridional_overturning_circulation
+        variables:
+          amoc:
+            mip: Omon
+            derive: true
```

Comment:
Also `derive` and `force_derivation` are fairly advanced features, not found in most recipes, so maybe should not be in a first example. It would be nice if you could add `start_year` and `end_year` in the variable section here and explain that to actually define a variable/dataset combination those keys are combined, with the keys from datasets taking precedence over the keys in variables.